src/test/java/com/google/sps/servlets/LiveCourseDataTest.java

```diff
@@ -0,0 +1,142 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public final class LiveCourseDataTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private AddSchoolData newSchoolObject;
+  private LiveCourseData LiveData;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    newSchoolObject = new AddSchoolData();
+    LiveData = new LiveCourseData();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+  @Mock HttpServletRequest requestB;
+
+  @Test
+  public void FindingExisitngTermEntity() {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request =
+        createRequest(
+            request, /*schoolName*/
+            "MIT", /*courseName*/
+            "6.006", /*termName*/
+            "Spring 2020", /*units*/
+            "12", /*profName*/
+            "Jason Ku");
+    requestB = createRequest(requestB, "MIT", "6.008", "Spring 2018", "6", "Srini");
+    String expectedTermName = "Spring 2020";
+
+    newSchoolObject.addSchoolData(db, request);
+    newSchoolObject.addSchoolData(db, requestB);
+    Key expectedParent = findQueryMatch(db, "Course", "course-name", "6.006").get(0).getKey();
+    Entity found = LiveData.getTerm(db, request);
+
+    assertEquals(expectedParent, found.getParent());
+    assertEquals(expectedTermName, found.getProperty("term"));
+  }
+
+  @Test
+  public void GettingRatingData_Hours_Difficulty() {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    Entity parent = addTermEntity(db);
+    addRatingEntity(db, 12, 7, parent.getKey());
```

Comment:
Annotating these numbers with what they're referring to can be helpful: `/* hours */ 12, /* difficulty */ 7`